### PR TITLE
[model, ops] feat: support aiter attention backend on ROCm

### PR DESCRIPTION
**Pull request by SkyHeroesS**
Adds `attn_implementation: aiter`, dispatching attention to AMD's aiter (AI Tensor Engine for ROCm https://github.com/ROCm/aiter) FMHA-v3 kernels as an alternative to flash-attn 2.8.3's composable-kernel build.

The registered name is `veomni_flash_attention_aiter_with_sp`, and it has to contain "flash": transformers' `is_flash_attention_requested()` dispatches on a plain `"flash" in impl` substring test, and a name without it sends the Omni audio/vision encoders down a non-varlen split path that size-mismatches under sequence parallelism.

Validated on 8xMI308X / ROCm 7.14 against `flash_attention_2` with the same seed and data. Wan2.1-1.3B DiT: 1.28x faster (5.42 -> 4.25 s/step), rising to 1.36x at real-480P latent size, per-step |delta| < 0.01 over 30 steps. Qwen3-Omni-MoE-30B: 1.03x on the default config, where packing makes attention only 7% of the step, and 1.13x -> 1.47x from 8k to 64k unpacked; max |delta| = 0.03 over 25 steps, the same as the FA2-vs-FA2 baseline. Full numbers in `docs/hardware_support/rocm/README.md`.

### What does this PR do?

> Concise overview of the change. Reference related issues/PRs.

### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> Validation results (training curves, eval metrics) for changes not covered by CI.

### API and Usage Example

> Show API changes and usage examples if applicable.

### Design & Code Changes

> High-level design description and specific change list.

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

## Summary by CodeRabbit

* **New Features**
  * Added AITER attention backend support for AMD ROCm systems.
  * Added sequence-parallel and variable-length attention support.
  * Added backend selection and model integration for AITER.
  * Added validation and actionable errors for missing installations and unsupported options.

* **Documentation**
  * Documented configuration, package requirements, compatibility, performance, and ROCm limitations.
  * Updated attention implementation and adapter guidance.

* **Tests**
  * Added coverage for registration, dispatch, validation, and supported attention behaviors.

### CODE REVIEW

File: veomni/ops/kernels/attention/aiter.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            q,
+            k,
+            v,
+            cu_seqlens_q=cu_seqlens_q,
+            cu_seqlens_k=cu_seqlens_k,
+            max_seqlen_q=max_seqlen_q,
+            max_seqlen_k=max_seqlen_k,
+            dropout_p=dropout_p,
+            softmax_scale=softmax_scale,
+            logits_soft_cap=float(softcap) if softcap else 0.0,
```
Comment (FoolPlayer) [inline veomni/ops/kernels/attention/aiter.py:180]: [P1] Could we reject nonzero softcap during training? In the documented AITER revision, forward applies `logits_soft_cap`, but backward neither saves nor receives it, producing incorrect gradients. Please add a forward/backward parity test before enabling this. [Upstream implementation](https://github.com/ROCm/aiter/blob/b5e03ed19/aiter/ops/mha.py#L2397).

File: veomni/ops/kernels/attention/aiter.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    aiter's kernels do take a ``sink_ptr``, so this is a candidate for real support later;
+    it is rejected rather than wired through because the semantics have not been validated
+    against the eager reference on ROCm.
+    """
+    if s_aux is not None or learnable_sink is not None:
+        raise ValueError(
+            "attn_implementation='aiter' does not support attention sinks yet (the model passed "
+            "s_aux/learnable_sink). Sink logits would be silently ignored, changing the attention "
+            "maths. Use attn_implementation='flash_attention_2' or 'eager' for sink models such as "
```
Comment (FoolPlayer) [inline veomni/ops/kernels/attention/aiter.py:69]: [P2] Could we recommend only `eager` here? As the ROCm docs note, FA2 2.8.3 silently drops attention sinks, so following this fallback suggestion would change the model's attention semantics.

File: docs/hardware_support/rocm/README.md [hunk 3]
```diff
@@ -101,10 +102,63 @@ The image bakes in the following ROCm-specific environment variables (you can ov
...
+[AITER](https://github.com/ROCm/aiter) is AMD's kernel library for ROCm. VeOmni can route attention to its FMHA-v3 kernels instead of the default flash-attn 2.8.3 composable-kernel build:
+
+```yaml
+model:
+  ops_implementation:
+    attn_implementation: aiter
+```
+
+`aiter` comes from the base image rather than being installed by the Dockerfile, so VeOmni does not
+pin it. The distribution is named `amd_aiter`, not `aiter`:
```
Comment (FoolPlayer) [inline docs/hardware_support/rocm/README.md:116]: Could we migrate the ROCm dependencies to a dedicated `rocm` extra (`uv sync --extra rocm`) instead of relying on the preinstalled image environment? What currently blocks this, and could we explicitly pin AITER and the ROCm stack there?

File: docs/hardware_support/rocm/README.md [hunk 4]
```diff
@@ -20,6 +20,7 @@ The image is based on `rocm/primus:v26.4` and ships a ROCm 7.14 stack tuned for
 | torchvision | `0.27.0+rocm7.14.0a20260608` |
 | triton | `3.7.0+gitb4e20bbe.rocm7.14.0a20260608` |
 | flash-attn | `2.8.3` |
+| aiter | `0.1.12.post2.dev214+gb5e03ed19` |
 | transformers | `5.12.1` |
```
Comment (FoolPlayer) [inline docs/hardware_support/rocm/README.md:24]: Which Transformers version was used for the reported validation? This image lists `5.12.1`, while current main pins `5.16.1`. Could we align the published image and docs with the repo pin and rerun the AITER checks on that version?
